service/history/execMgrFactory.go

```diff
@@ -56,12 +56,5 @@ func (factory *executionMgrFactory) CreateExecutionManager(shardID int) (persist
 		shardID,
 		factory.logger)
 
-	if err != nil {
```

Comment:
If we are unable to create the persistence client then there is no need to create the metric client wrapper.  Just return nil with the error like we used to before.